src/ts/background/handlers/Extension.ts

```diff
@@ -50,69 +56,89 @@ export default class Extension {
     return true
   }
 
-  private authorizeApprove ({ id }: MessageAuthorizeApprove): boolean {
+  private authorizeApprove ({ id }: RequestAuthorizeApprove): boolean {
     const queued = this.state.getAuthRequest(id)
     assert(queued, 'Unable to find request')
     const { resolve } = queued
     resolve(true)
     return true
   }
 
-  private authorizeReject ({ id }: MessageAuthorizeReject): boolean {
+  private authorizeReject ({ id }: RequestAuthorizeReject): boolean {
     const queued = this.state.getAuthRequest(id)
     assert(queued, 'Unable to find request')
     const { reject } = queued
     reject(new Error('Rejected'))
     return true
   }
 
-  private signingApprove ({ id, password }: MessageExtrinsicSignApprove): boolean {
+  private signingApprovePassword ({ id, password }: RequestSigningApprovePassword): boolean {
     const queued = this.state.getSignRequest(id)
     assert(queued, 'Unable to find request')
     const { request, resolve, reject } = queued
-    if (!keyringVault.accountExists(request.address)) {
+    if (!keyringVault.accountExists(request.inner.address)) {
       reject(new Error('Unable to find account'))
       return false
     }
-    const payload = createType('ExtrinsicPayload', request, { version: request.version })
     keyringVault.unlock(password).then(() => {
-      const pair = keyringVault.getPair(request.address)
-      const signature = payload.sign(pair)
+      const inner = request.inner
+      const pair = keyringVault.getPair(inner.address)
+      let registry
+      if ((inner as SignerPayloadRaw).data) {
+        registry = new TypeRegistry()
+      } else {
+        const signerPayload = (inner as SignerPayloadJSON)
+        const network = findNetwork(signerPayload.genesisHash)
+        registry = network.registry
+      }
+      const result = request.sign(registry, pair)
       resolve({
         id,
-        ...signature
+        ...result
       })
     })
     return true
   }
 
-  private signingCancel ({ id }: MessageExtrinsicSignCancel): boolean {
+  private signingApproveSignature ({ id, signature }: RequestSigningApproveSignature): boolean {
+    const queued = this.state.getSignRequest(id)
+    assert(queued, 'Unable to find request')
+    const { resolve } = queued
```

Comment:
we haven't implemented the logic here, right?

Reply:
Actually this is implementation. It resolves to a `ResponseSigning` object - `{ id, signature }`. The signature is passed from popup instead of generated in background as `signingApprovePassword` implementation. In the popup, `Qr` is the component that presents a QR code for user to scan using mobile apps such as Parity Signer.